org.eclipse.draw2d/src/org/eclipse/draw2d/Container.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Johannes Kepler University Linz
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Alois Zoitl - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.draw2d;
+
+/**
+ * Lightweight Container which just draws the children according to the given
+ * layout.
+ *
+ * This Container does not maintain any graphics state and delegates that to its
+ * children. This can save memory and increase drawing performance for larger
+ * graphs with deep nested figures.
+ *
+ * A Container can not have any border and does not draw any background.
+ *
+ * @since 3.15
+ */
+public class Container extends Figure {
+
+	public Container(LayoutManager manager) {
+		setLayoutManager(manager);
```

Comment:
I'm not a huge fan of calling public methods inside the constructor, because subclasses could easily overwrite `setLayoutManager()` and cause a whole lot of undesired side-effects on a partially initialized object.

What do you think about making this method final as well?
i.e. by adding:

```
@Override
public final void setLayoutManager(LayoutManager manager) {
    super.setLayoutManager(manager);
}
```

Reply:
I also had planed to experiment using this class in different places in 4diac IDE and gradually refine it. Can/should I flag this class as experimental for the next release so that we still can fine tune its API?

Reply:
> Originally I wanted to make the whole class final. However when searching in GEF Classic for use cases I noticed that there are use cases for subclassing. Therefore making this method final is a good idea.

I also like the idea a lot. I can already see it in the WindowBuilder project to represent e.g. the GridLayout or BorderLayout of a Composite.

> Can/should I flag this class as experimental for the next release so that we still can fine tune its API?

I don't know of a "proper" way to do this but the two options I'm aware of is to move this class to a separate package which is then exported as "provisional API", or to use things like the `@noreference` annotation.
See: https://wiki.eclipse.org/PDE/API_Tools/Restrictions

But in its current state, I think it's safe to keep it like that. You can generally overwrite or add new methods without breaking binary compatibility.

Reply:
This sounds a bit over complicated. I only feared that we may want to make more methods final. But we have 4 more weeks to test and I plan to start using it in 4diac IDE as soon as it is merged. So I guess you are right we should be save.

Reply:
Alternatively, you can always just keep it in 4diac for the time being and merge it into GEF, once you've refined the class.